env/env_encryption.cc

```diff
@@ -61,10 +59,8 @@ Status EncryptionProvider::CreateFromString(
   }
   if (id == kCTRProviderName) {
     result->reset(new CTREncryptionProvider());
-  } else if (StartsWith(id, kIPPAESProviderName)) {
-    IppAESCTRProvider* ipp_provider = nullptr;
-    status = IppAESCTRProvider::CreateProvider(id, &ipp_provider);
-    result->reset(ipp_provider);
+  } else if (id == kIPPAESProviderName) {
```

Comment:
The way I have done this for other classes is to do the following in your provider class:
static const char *kName() { return "IPP_AES"; }
const char *Name() const override ( return kName(); }

Then this method can use StartsWith(id, IppAESCTRProvider::kName())